src/instrument.mli

```diff
@@ -0,0 +1,14 @@
+open Import
+
+type pos = Before | After
+
+type t = { transformation :  Parsetree.structure -> Parsetree.structure ; position : pos}
```

Comment:
The type is abstract in `ppxlib.ml`. It's not abstract here since I need to access the fields in `driver.ml`. 

As an alternative, I could not have a `instrumentor.mli`-file at all and only define the signature of it in `ppxlib.ml`. Or I could also make `t` abstract in `instrumentor.mli` and define getters there that I wouldn't expose in `ppxlib.ml`. Does any of those sound better to you?

Reply:
I could also include this very small module into the driver module and have the type abstract from the beginning on and include it into `ppxlib.ml` as `module Instrumentor = Driver.Instrumentor`.

Reply:
When we can, it is better to keep the internal and external interfaces the same. It makes it easier to understand the code as we don't need to remember that `ppxlib.ml` changes the API. It also plays better with tools as well, such as documentation generators or merlin. Here, if someone just to the `Instrumentor.t` type via merlin they'll reach the version in ppxlib.ml which is not documented.

In this case, either adding getters or as a sub-module of `driver.mli` seems fine. Given that it is a small module that is likely to only be used at the registration stage, I would personally put it in `Driver` and not even add the `module Instrumentor = Driver.Instrumentor` in `ppxlib.ml`. This is in line with what is done for `Cookies` for instance.

Reply:
Sounds convincing to me. I've made a sub-module of `Driver` out of it.